src/components/Layout.js

```diff
@@ -1,54 +1,140 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import Sidebar from './Sidebar';
 import { GiHamburgerMenu } from 'react-icons/gi';
+import { FaExclamationTriangle, FaTimes } from 'react-icons/fa'; // Import the icons you want to use
 import logo from '../assets/images/wallet_white.png';
 import { useLocation, useNavigate } from 'react-router-dom';
 
-const Layout = ({ children }) => {
-	const location=useLocation();
-	const navigate = useNavigate();
-
-	const [isOpen, setIsOpen] = useState(false);
-
-	const toggleSidebar = () => setIsOpen(!isOpen);
-
-	const handleNavigate = (path) => {
-		if (location.pathname === path) {
-			window.location.reload();
-		} else {
-		navigate(path);
-		// toggle(); // Close the sidebar after navigation (optional)
-		}
-	};
-
-	return (
-		<div className="flex h-screen">
-			<Sidebar isOpen={isOpen} toggle={toggleSidebar} />
-
-			{/* Header */}
-			<header className={`${isOpen ? 'hidden' : 'z-50 fixed top-0 left-0 w-full bg-custom-blue text-white flex items-center justify-between p-4 sm:hidden'}`}>
-				<div className="flex items-center">
-					<img src={logo} alt="Logo" className="w-10 h-auto mr-2 cursor-pointer" onClick={() => handleNavigate('/')}/>
-				</div>
-				<h1 className="text-white text-xl font-bold cursor-pointer" onClick={() => handleNavigate('/')} >wwWallet</h1>
-				<button className="text-white" onClick={toggleSidebar}>
-					<GiHamburgerMenu size={24} />
-				</button>
-			</header>
-
-			<div className="w-3/5 flex flex-col flex-grow">
-				{/* Sidebar */}
-				<div className={`sticky top-0 h-screen overflow-y-auto bg-custom-blue text-white p-6 sm:w-64 ${isOpen ? 'block' : 'hidden'}`}>
-					<Sidebar isOpen={isOpen} toggle={toggleSidebar} />
-				</div>
-
-				{/* Content */}
-				<div className="flex-grow bg-gray-100 p-6 mt-10 pt-10 sm:mt-0 sm:pt-6 overflow-y-auto">
-					{children}
-				</div>
-			</div>
-		</div>
-	);
+const Layout = ({ children, isPermissionGranted, isPermissionValue,setispermissionValue }) => {
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const [isOpen, setIsOpen] = useState(false);
+  const [isMessageVisible, setIsMessageVisible] = useState(true);
+  const toggleSidebar = () => setIsOpen(!isOpen);
+
+  const handleNavigate = (path) => {
+    if (location.pathname === path) {
+      window.location.reload();
+    } else {
+      navigate(path);
+    }
+  };
+
+  const handleCloseMessage = () => {
+    setIsMessageVisible(false);
+    // Store the isMessageVisible state in session storage
+    sessionStorage.setItem('isMessageVisible', 'false');
```

Comment:
Have you tried using the `useSessionStorage("isMessageVisible", false)` hook from the `useStorage` module? This hook behaves like the `useState` hook, and automatically re-renders the component if the stored state changes through any other hook instance with the same name. You use it just like you would use `useState`:

```js
  const [isMessageVisible, setIsMessageVisible] = useSessionStorage("isMessageVisible", true);
```

(See #131 for a fix of one caveat, though)

Reply:
I plan to update this suggestion in a separate pull request, which will include too the enhancement of avoiding unnecessary backend posts for storing FCM tokens